packages/hash/frontend/src/components/hooks/blockProtocolFunctions/useBlockProtocolCreateLinks.ts

```diff
@@ -39,37 +52,59 @@ export const useBlockProtocolCreateLinks = (): {
         };
 
         if ("operation" in action) {
-          throw new Error(
-            "Creating new linkedAggregations not yet implemented.",
+          if (!action.operation.entityTypeId) {
+            throw new Error(
+              "entityTypeId is compulsory on operation while trying to create a linkedAggregation",
+            );
+          }
+
+          const { data, errors } = await runCreateLinkedAggregationMutation({
+            variables: {
+              ...action,
+              operation: {
+                // @todo this shouldn't be necessary
+                ...action.operation,
+                entityTypeId: action.operation.entityTypeId,
+              },
+              sourceAccountId: action.sourceAccountId,
+            },
+          });
+          if (!data) {
+            throw new Error(`Could not create link: ${errors?.[0]!.message}`);
+          }
+
+          // @todo, add a proper typecheck. The GraphQL query for multiFilter { operator } returns String, but BlockProtocolLinkedAggregationUpdateMutationResults defines the exact type for operator. This typecast is used to typecast string to the one the query expects.
+          results.push(
+            data.createLinkedAggregation as unknown as BlockProtocolLink,
           );
         } else if (!("destinationAccountId" in action)) {
           throw new Error(
             "One of operation or destinationEntityId must be provided",
           );
-        }
+        } else {
```

Comment:
This part remains the same. It has been wrapped in an `else` block for better clarity and type-checking.